U2E/src/main/java/Konkuk/U2E/domain/comment/service/CommentService.java

```diff
@@ -0,0 +1,32 @@
+package Konkuk.U2E.domain.comment.service;
+
+import Konkuk.U2E.domain.comment.dto.response.CommentInfo;
+import Konkuk.U2E.domain.comment.dto.response.GetCommentsResponse;
+import Konkuk.U2E.domain.comment.repository.CommentRepository;
+import Konkuk.U2E.domain.news.exception.NewsNotFoundException;
+import Konkuk.U2E.domain.news.repository.NewsRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import static Konkuk.U2E.global.response.status.BaseExceptionResponseStatus.NEWS_NOT_FOUND;
+
+@Service
+@RequiredArgsConstructor
+public class CommentService {
+    private final CommentRepository commentRepository;
+    private final NewsRepository newsRepository;
+
+    public GetCommentsResponse getCommentInfo(Long newsId) {
+        newsRepository.findById(newsId)
+                .orElseThrow(() -> new NewsNotFoundException(NEWS_NOT_FOUND));
+
+        return GetCommentsResponse.of(commentRepository.findCommentsByNewsId(newsId).stream()
+                .map(c -> new CommentInfo(
+                        c.getUser().getUserId(),
+                        c.getUser().getName(),
+                        c.getCommentId(),
+                        c.getContents()
+                )).toList()
+        );
```

Comment:
`CommentInfo`도 정적 팩토리 메서드를 사용하면 코드가 조금더 깔끔해질 것 같습니다!

```suggestion
        return GetCommentsResponse.of(commentRepository.findCommentsByNewsId(newsId).stream()
                .map(c -> CommentInfo.of(c))
                .toList()
        );
```

</br>
아래처럼 메서드 참조도 가능할 것 같네요.

```java
return GetCommentsResponse.of(commentRepository.findCommentsByNewsId(newsId).stream()
                .map(c -> CommentInfo::of)
                .toList()
        );
```

Reply:
오 훨씬 깔끔하고 가독성도 좋네요! 바로 수정하겠습니다!